eng/common/cross/toolchain.cmake

```diff
@@ -218,7 +226,7 @@ endif()
 
 # Specify compile options
 
-if((TARGET_ARCH_NAME MATCHES "^(arm|armv6|armel|arm64|ppc64le|s390x)$" AND NOT ANDROID) OR ILLUMOS)
+if((TARGET_ARCH_NAME MATCHES "^(arm|armv6|armel|arm64|ppc64le|s390x)$" AND NOT ANDROID AND NOT FREEBSD) OR ILLUMOS)
```

Comment:
Becuase, when building from Linux, TOOLCHAIN will be set to `aarch64-linux-gnu` (line 52) when targeting arm64, then without this `AND NOT FREEBSD` it will overwrite `CMAKE_C..._COMPILTER_TARGET` (line 230 to 232)  and build will fail.

Reply:
I am not sure how we would get to line 52 with your other changes. It seems that we should end up at line 57 instead and then I was thinking that we could set the proper TOOLCHAIN instead of the triple.
But now I can see that we do something similar for x64 FreeBSD too. So maybe there is some additional reason for that that I don't see.